### DISCUSSION TITLE
[Feature] Multi-server setup

### DISCUSSION THREAD

**Discussion by Szumi-github** [immich-app/immich#11850] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi there!
First of all, I love the app, and huge kudos to all the developers for creating such amazing software.

My setup: I have a Raspberry Pi 4 (with a 4TB HDD drive connected to it) on which I run my Immich through docker-compose.

What I want to accomplish: I want to set up my second RPi 4 in a different location and treat it as a backup server. So whenever the first server goes dark (power goes off or the internet) I want to be able to switch (by using for example Traefik) to my second server so the Immich app is always on and up to date. 

My question: Question is, how can I set up my backups/synchronization so that my second server is always up to date, and also whenever the first server comes back alive, it syncs what has changed in the meantime? I know that a backup of the database can be done while Immich is running so I can automate it, but what about folders with the images? Right now as it is described in the documentation, I need to turn off the Immich to backup my files.

I was thinking about setting up Syncthing to sync my files but then the Immich database would not update and everything would break. 

Is there any solution for this? or maybe there is some common approach for dealing with multi-server synchronization?

Huge thanks for all the responses in advance!

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by C-Otto** [comment]
I actually did something like that once, for a commercial project. You can synchronize the disks using DRBD (https://linbit.com/drbd/) and deal with the up/down issues using Corosync and Pacemaker. This guide seems promising: https://houseofbrick.com/blog/active-passive-cluster-for-near-ha-using-pacemaker-drbd-corosync-and-mysql/

If you want to do it right, it's a lot of work, though. File-system snapshots (ZFS?) might be good enough for your use case - but that's still quite a bit of work.

Note that most high-availability solutions DO NOT work if there are only two servers which cannot communicate. Due to the split brain issue, each side cannot determine if it should take over. You either need to come up with a majority (2 out of 3), or ensure that the other side is down using the funny concept of STONITH: Shoot The Other Node In The Head (for example by turning off its power - if you can due that despite lack of internet connectivity).

**Comment by bo0tzz** [comment]
This has been requested a few times now. Doing it correctly is (very) hard and requires serious ops knowledge - and most people with the necessary knowledge would probably decide to just not do it ;) Either way, it's definitely not in scope for Immich to manage.
